### PR TITLE
Fix ignore_changes behavior for AGW

### PR DESCRIPTION
**Pull request by kvendingoldo**

### CODE REVIEW

**Comment by ztbrown** [comment]
this looks good, but I would like to see a test. That's what was keeping me from PRing.

**Comment by kvendingoldo** [comment]
@ztbrown Will try to do it today or tomorrow

**Comment by ztbrown** [comment]
unit / acceptance was what I was thinking

**Comment by kvendingoldo** [comment]
@ztbrown I created a draft of unit test that should work for us. For now, I need to find a way how to change AGW certificate via Go code, after that it could be seen as a whole change.

File: azurerm/internal/services/network/tests/application_gateway_resource_test.go [hunk 1]
```diff
@@ -4011,6 +4040,207 @@ resource "azurerm_application_gateway" "test" {
...
+		if err != nil {
+			return fmt.Errorf("Bad: updating AGW: %+v", err)
+		}
+
+		if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("Bad: waiting for update of AGW: %+v", err)
+		}
+
+		return nil
+
```
Comment (ztbrown) [inline azurerm/internal/services/network/tests/application_gateway_resource_test.go:4082]: remove this newline

**Comment by ztbrown** [comment]
Nice work! Did you see that you're failing the linter? 

image

Need to remove the newline on 4082.

**Comment by kvendingoldo** [comment]
@ztbrown I've seen and will fix it a little bit later. For now, I'm not able to fix my test, have the following issue:

```
    TestAccAzureRMApplicationGateway_manualSslCertificateChangeIgnoreChanges: testing.go:684: Step 0 error: Check failed: Check 3/4 error: Bad: updating AGW: network.ApplicationGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidResourceReference" Message="Resource /subscriptions/XXX/resourceGroups/acctestRG-201008145214600822/providers/Microsoft.Network/applicationGateways/acctesttag/sslCertificates/acctestcertificate1 referenced by resource /subscriptions/XXX/resourceGroups/acctestRG-201008145214600822/providers/Microsoft.Network/applicationGateways/acctesttag/httpListeners/acctest-vnet-201008145214600822-httplstn was not found. Please make sure that the referenced resource exists, and that both resources are in the same region." Details=[]
```
Do you have any idea why `agw.SslCertificates = &newSslCertificates` isn't working fine?

File: azurerm/internal/services/network/tests/application_gateway_resource_test.go (hunk 1 above)
Comment (kvendingoldo) [reply azurerm/internal/services/network/tests/application_gateway_resource_test.go]: why?

Comment (ztbrown) [reply azurerm/internal/services/network/tests/application_gateway_resource_test.go]: I believe this is what is upsetting the linter.

**Comment by kvendingoldo** [comment]
Screen Shot 2020-10-09 at 3 21 44 AM

**Comment by kvendingoldo** [comment]
@ztbrown probably I finished with tests. For now I need to find a way of fixing linter issues and after that it could be reviewed and merged

**Comment by ztbrown** [comment]
👍 Looking great!

**Review by ztbrown — APPROVED** [review]

**Comment by kvendingoldo** [comment]
@katbyte @mbfrahry @tombuildsstuff can you check it and merge, pls?

**Comment by kvendingoldo** [comment]
@manicminer @jackofallops can you check it, please?

**Review by jackofallops — APPROVED** [review]
Thanks @kvendingoldo - This LGTM 👍 
Tests Passing:
image

**Pull request merged**